src/main/js/src/store/reducers/registration.js

```diff
@@ -12,23 +15,54 @@ const initialState = {
 
 const reducer = (state = initialState, action) => {
   switch (action.type) {
-    case actionTypes.FETCH_EXAM_LOCATIONS_START:
+    case actionTypes.FETCH_EXAM_SESSIONS_START:
       return {
         ...state,
         loading: true,
       };
-    case actionTypes.FETCH_EXAM_LOCATIONS_SUCCESS:
+    case actionTypes.FETCH_EXAM_SESSIONS_SUCCESS:
       return {
         ...state,
         loading: false,
-        locations: action.locations,
+        examSessions: action.examSessions,
       };
-    case actionTypes.FETCH_EXAM_LOCATIONS_FAIL:
+    case actionTypes.FETCH_EXAM_SESSIONS_FAIL:
       return {
         ...state,
         loading: false,
         error: action.error,
       };
+    case actionTypes.FILTER_AND_GROUP_BY_DATE:
```

Comment:
Consider using ramda for data transformations. For example group by session date would be:
R.groupBy(s => s.session_date, filteredByLanguage)

Reply:
You are totally correct.
I need to start looking into Ramda and find out what it can do.